wls-gui-wahllokalsystem/src/stores/ereignisStore.ts

```diff
@@ -16,23 +21,57 @@ export const useEreignisStore = defineStore(storeID, () => {
   const error = ref<string | null>(null);
 
   const { currentUserWahlbezirkID } = storeToRefs(useUserStore());
+  const { schliessungsUhrzeitSent } = storeToRefs(useWahlbezirkStore());
+
   const wahlbezirkEreignisse = ref<WahlbezirkEreignisse>(
     WahlbezirkEreignisseBuilder.createEmptyWahlbezirkEreignisse()
   );
 
+  const hasEintraege = computed(
+    () => (wahlbezirkEreignisse.value.ereigniseintraege ?? []).length > 0
+  );
+  const hasVorfaelle = computed(
+    () =>
+      wahlbezirkEreignisse.value.ereigniseintraege?.some(
+        (eintrag) => eintrag.ereignisart === EreignisartEnum.Vorfall
+      ) === true
+  );
+  const hasVorkommnisse = computed(
+    () =>
+      wahlbezirkEreignisse.value.ereigniseintraege?.some(
+        (eintrag) => eintrag.ereignisart === EreignisartEnum.Vorkommnis
+      ) === true
+  );
+  const areKeineEreignisseFlagsValid = computed(
+    () =>
+      hasVorfaelle.value !== wahlbezirkEreignisse.value.keineVorfaelle &&
+      (hasVorkommnisse.value !== wahlbezirkEreignisse.value.keineVorkommnisse ||
+        !schliessungsUhrzeitSent.value)
+  );
+
+  watch(schliessungsUhrzeitSent, _onSchliessunguhrzeitSentChanged);
+  watchEffect(() => _updateKeineFlagsOfEreignisseBasedOnCurrentState());
+
   function addEreignis() {
+    const currentDate = new Date();
+    const newEreignisEreignisart =
```

Comment:
warum `EreignisEreignisart`? 
```suggestion
    const newEreignisart =
```

Reply:
`newEreignisart ` war mir zu generisch. Ich wollte zum Ausdruck bringen dass es die ereignisart des neues Ereignisses ist. Wäre `ereignisartOfNewEreignis` oder `..OfEreignisToAdd` besser?

Reply:
dass es vom neuen Ereignis ist, ist doch klar durch den Kontext dass wir uns in `addEreignis()` befinden. 

alternativ könte man es auch so machen:
```
function addEreignis() {
    wahlbezirkEreignisse.value.ereigniseintraege?.push({
      uhrzeit: new Date(),
      beschreibung: "",
      ereignisart: getEreignisArtForDateRelatedToSchliessungsuhrzeit(
        currentDate,
        schliessungsUhrzeitSent.value
      ),
    });
  }
  ```

Reply:
fixed mit e465e064deecda74da3db08c1b3c478c51bebe65